### DISCUSSION TITLE
Chat with suggestions

### DISCUSSION THREAD

**Discussion by ichoosetoaccept** [cline/cline#408] [Feature Requests]
Something I think would be very cool is the ability to chat with code suggested by Claude Dev.

The reasoning behind this is that that suggested code may involve advanced language feature the user is not aware of or has never seen before. Even if the user does understand everything in a suggestion, he may have questions about the particular suggestion being made. It would be nice to be able to "pause" the flow of a Claude Dev task so the user could ask questions like 
- why did you add this here?
- what purpose does it serve?
- does this align with best practices?
etc.

In a sense this would spawn a separate, unrelated conversation about the changes happening to the code and would greatly enhance the user's learning of a language, if he or she wishes to make use of this.

Taking this further, we could make the choice of LLM user-configurable for these separate conversations, providing even more flexibility.

**Comment by TomLucidor** [comment]
I think this is very important but isn't this available already? Or when it comes to goal alignment, should it have something like CursorRules for helping with these kind of things?

**Reply by lloydchang** [reply]
as an alternative, you can try a fork https://github.com/RooVetGit/Roo-Cline/commit/a6bfbf8bcb4b27ae02a8ae35c4b0fdd5a5417416#diff-ec5bfb97fa984921dcb420c3fcf9747854c3302ac5ec437f1fe604ee68fc4e7aR287
> `const ruleFiles = ['.clinerules', '.cursorrules']`

and its `.vsix` at https://marketplace.visualstudio.com/items?itemName=RooVeterinaryInc.roo-cline

from https://discord.com/channels/1275535550845292637/1275535550845292640/1312113711842066524

@mrubens wrote:
> In the fork I've been working on I have Cline reading .cursorrules and .clinerules from the working directory and interpolating them into the system prompt if present (at least in our use case most of the things we tell Cursor are also relevant for Cline). 

related https://github.com/cline/cline/discussions/361 https://github.com/cline/cline/discussions/408 https://github.com/cline/cline/discussions/622
